conjur-api/Variable.cs

```diff
@@ -47,17 +48,37 @@ public string GetValue()
         /// </summary>
         /// <param name="val">Secret value.</param>
         public void AddSecret(string val)
+        {
+            byte[] data = Encoding.UTF8.GetBytes(val);
+            AddSecret(data);
+        }
+
+        /// <summary>
+        /// Set a secret (value) to this variable.
+        /// The value is being cleared after used!!!
+        /// </summary>
+        /// <param name="val">Secret value.</param>
+        public void AddSecret(byte[] val)
         {
             WebRequest webRequest = this.Client.AuthenticatedRequest(this.path);
             webRequest.Method = "POST";
+#if (SIGNING)
```

Comment:
Please check if we can inherit our MockRequest from HttpWebRequest instead of WebRequest. If so we can to eliminate this `#if`

Reply:
Instead of inherit use the
if (webRequest is HttpWebRequest)